Lib/test/libregrtest/main.py

```diff
@@ -248,6 +250,31 @@ def list_tests(self):
         for name in self.selected:
             print(name)
 
+    def _list_cases(self, suite):
+        for test in suite:
+            if isinstance(test, unittest.loader._FailedTest):
+                continue
+            elif isinstance(test, doctest.DocTestCase):
+                continue
```

Comment:
Please remove this if: doctest falls into the the unittest.TestCase case, and... it just works! I'm able to pass to output to -m test --matchfile=tests and only these doctest tests are executed!